### DISCUSSION TITLE
You can send multiple files with adding more **`-F `**

### DISCUSSION THREAD

**Discussion by saireddy12** [fastapi/fastapi#6984] [Questions] [answered]
You can send multiple files with adding more **`-F `**

```shell
curl -X POST "http://127.0.0.1:8000/uploadfile" -H  "accept: application/json" -H  "Content-Type: multipart/form-data" \
 -F "input_data=@two.png;type=image/png" -F "input_data=@three.png;type=image/png"
```

_Originally posted by @ycd in https://github.com/tiangolo/fastapi/issues/1856#issuecomment-676291178_

**Comment by saireddy12** [comment]
Is there a way to do this in python. I need to hit the api with multiple images 
```
@app.post("/text")
def get_text(files: List[UploadFile] = File(...),frame_timestamps: Optional[List[int]] = Body([]),is_video:bool=False)
```
its working when I try uploading multiple images using /docs interface, I tried with one file its working fine here is the code for it 
```
import requests
import json
def get_text(image_path):
    #images={}
    url = 'http://address/text''
    try:
        with open(image_path, "rb") as im:
            image_data={"files":im}
            response=requests.post(url,files=image_data)
            return json.loads(response.text)
    except Exception as er:
        print("error occured")
        return "{} error occured".format(er)
```
when I tried adding one more image. to the image_data I am getting error . tried this 
```
image_data ={"files":[]}
for image in image_list:
    with open(image, "rb") as im:
        image_data['files'].append(im)
```

**Comment by Kludex** [comment]
What error?

`with` statement missing btw

**Comment by saireddy12** [comment]
> What error?
> 
> `with` statement missing btw

I have updated the question and I am attaching the error message screenshot here

Thanks..

<img width="1015" alt="Screenshot 2021-01-25 at 9 27 02 AM" src="https://user-images.githubusercontent.com/38287214/105659926-e3faef00-5eef-11eb-93c5-d46bfb9b1f26.png">

**Comment by ycd** [comment]
You are getting the error on client side, see it directly comes from `/site-packages/requests/models.py`

**Comment by ycd** [comment]
Try this one:

```python
files = {files: [open('image1.jpg', 'rb'), open('image2.jpg', 'rb')]}
response=requests.post(url,files=files)
```

**Comment by saireddy12** [comment]
> Try this one:
> 
> ```python
> files = {files: [open('image1.jpg', 'rb'), open('image2.jpg', 'rb')]}
> response=requests.post(url,files=files)
> ```
No use

here is the error message 
<img width="1015" alt="Screenshot 2021-01-25 at 2 41 48 PM" src="https://user-images.githubusercontent.com/38287214/105685020-a57b2980-5f1b-11eb-9cc1-1f39eefa9f96.png">

**Comment by ycd** [comment]
Can you try converting this to `files: List[UploadFile] = File(...),` => `files: List[bytes] = File(...),`, then read image as bytes like we did above, pass them in the here again.

```python
myimage1 = open('image1.jpg', 'rb')
myimage2 = open('image2.jpg', 'rb')
files = {files: [myimage1, myimage2]}
response=requests.post(url, files=files)
```

**Comment by saireddy12** [comment]
thanks 

> Can you try converting this to `files: List[UploadFile] = File(...),` => `files: List[bytes] = File(...),`, then read image as bytes like we did above, pass them in the here again.
> 
> ```python
> myimage1 = open('image1.jpg', 'rb')
> myimage2 = open('image2.jpg', 'rb')
> files = {files: [myimage1, myimage2]}
> response=requests.post(url, files=files)
> ```

Thanks for reply

I tried as you said, but got same error , its working fine If I upload multiple images using localhost:/docs

Here is the screenshot of the error
<img width="1104" alt="Screenshot 2021-01-27 at 5 07 10 PM" src="https://user-images.githubusercontent.com/38287214/105986038-5ae7e180-60c2-11eb-8fcb-a971f4858c48.png">

Thanks..

**Comment by saireddy12** [comment, marked as answer]
@ycd @Kludex Thanks for the help,

 I finally found the solution , its not the problem of fastapi , its related to requests library 

Incase any one  needs  the solution here it is 
```
files = [
        ('files', ('image1', open('/Users/ai/image1.jpg','rb'), 'image/png')),
        ('files', ('image2', open('/Users/ai/image2.jpeg','rb'), 'image/png'))
        ]
```
you can use the below function for multiple files
```
import requests
import json

def get_text(image_list,url):
    try:
        image_data=[]
        for image in image_list:
            image_data.append(('files',(image.split('/')[-1],open(image,'rb'),'image/png')))#('files',(image_name,open image,type))
        response=requests.post(url,files=image_data)
        return json.loads(response.text)
    except Exception as er:
        print("error occured")
        return "{} error occured".format(er)
```

You can check docs [Here][1] 

Thanks..!


  [1]: https://requests.readthedocs.io/en/latest/user/advanced/#post-multiple-multipart-encoded-files

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @saireddy12  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
